### PR TITLE
[JENKINS-30351] Configuration option to write plain timestamp entries on build log file

### PR DESCRIPTION
**Pull request by amuniz**
[JENKINS-30351](https://issues.jenkins-ci.org/browse/JENKINS-30351)

### CODE REVIEW

**Comment by amuniz** [comment]
@reviewbybees

File: src/main/java/hudson/plugins/timestamper/PlainTimestampNotesOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Output stream that writes each line to the provided delegate output stream
+ * after inserting a {@link TimestampNote} as plain text.
+ */
+class PlainTimestampNotesOutputStream extends TimestampNotesOutputStream {
+
+  PlainTimestampNotesOutputStream(OutputStream delegate) {
```
Comment (tfennelly) [inline src/main/java/hudson/plugins/timestamper/PlainTimestampNotesOutputStream.java]: `@Nonnull` ?

File: src/main/java/hudson/plugins/timestamper/TimestampNote.java [hunk 2]
```diff
@@ -104,4 +113,11 @@ public Timestamp getTimestamp(Run<?, ?> build) {
     }
     return null; // each time-stamp note affects one line only
   }
+
+  public void plain(OutputStream out) throws IOException {
+    TimestamperConfig config = GlobalConfiguration.all().get(TimestamperConfig.class);
```
Comment (tfennelly) [inline src/main/java/hudson/plugins/timestamper/TimestampNote.java]: Can this not be retrieved once (statically) and avoid the lookup for every log? All that stuff is static anyway.

File: src/main/java/hudson/plugins/timestamper/TimestampNote.java [hunk 3]
```diff
@@ -63,6 +68,10 @@ public static String getSystemProperty() {
     return "timestamper-consolenotes";
   }
 
+  public static String getUsePlainSystemProperty() {
+    return "timestamper-consolenotes-plain";
```
Comment (tfennelly) [inline src/main/java/hudson/plugins/timestamper/TimestampNote.java]: Why not just defined this as a `public static final String` Vs a method?

**Comment by tfennelly** [comment]
:bee: 

Would it make sense to document this somewhere i.e. how to configure the option? Maybe in the plugin's `README.md`, or on the wiki page.

File: src/main/java/hudson/plugins/timestamper/TimestampNote.java (hunk 2 above)
Comment (amuniz) [inline src/main/java/hudson/plugins/timestamper/TimestampNote.java]: What would happen if the user change the configuration? I think we won't see the changes.

File: src/main/java/hudson/plugins/timestamper/TimestampNote.java (hunk 3 above)
Comment (amuniz) [inline src/main/java/hudson/plugins/timestamper/TimestampNote.java]: Just to follow the existent "convention" in this plugin (see the above method)

**Comment by jglick** [comment]
Questioned the motivation in JIRA.

**Comment by StevenGBrown** [comment]
I'm sorry, I cannot accept this pull request.

More discussion in the JIRA.

**Pull request closed without merging**